spark-connect/client/src/main/scala-spark-master/io/delta/connect/tables/DeltaTable.scala

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.tables
+
+import scala.collection.JavaConverters._
+
+import io.delta.connect.proto
+import io.delta.connect.spark.{proto => spark_proto}
+
+import org.apache.spark.sql.{DataFrame, Dataset, Row, SparkSession}
+import org.apache.spark.sql.connect.delta.ImplicitProtoConversions._
+
+/**
+ * Main class for programmatically interacting with Delta tables.
+ * You can create DeltaTable instances using the static methods.
+ * {{{
+ *   DeltaTable.forPath(sparkSession, pathToTheDeltaTable)
+ * }}}
+ *
+ * @since 3.3.0
+ */
+class DeltaTable private[tables](
+    private val df: Dataset[Row],
+    private val table: proto.DeltaTable)
+  extends Serializable {
+
+  private def sparkSession: SparkSession = df.sparkSession
+
+  /**
+   * Apply an alias to the DeltaTable. This is similar to `Dataset.as(alias)` or
+   * SQL `tableName AS alias`.
+   *
+   * @since 3.3.0
```

Comment:
Wasn't this introduced earlier than 3.3.0?

Reply:
@allisonport-db @tdas even though the functions, classes and objects are similar to [here](https://github.com/delta-io/delta/blob/master/spark/src/main/scala/io/delta/tables/DeltaTable.scala#L44), they have different implementations, and one uses Spark Connect while the other uses the normal Spark, the @since of these new functions should start from 3.3.0 right? (3.2.0 has been releasted, so I'm assuming it is 3.3.0?)

Reply:
This is kind of a weird scenario since they're the same APIs. I think it should be since `4.0.0`... we won't be releasing these on `3.x.x` since it needs Spark 4.0.

Reply:
This makes me think though that we do want to publish these Java docs?

Reply:
@allisonport-db Why do we want to publish these Java docs at the moment?